app/src/main/java/org/mozilla/fenix/settings/HomeSettingsFragment.kt

```diff
@@ -116,6 +126,44 @@ class HomeSettingsFragment : PreferenceFragmentCompat() {
             }
         }
 
+        requirePreference<CheckBoxPreference>(R.string.pref_key_pocket_sponsored_stories).apply {
+            isVisible = FeatureFlags.isPocketSponsoredStoriesFeatureEnabled(context)
+            isChecked = context.settings().showPocketSponsoredStories
+            onPreferenceChangeListener = object : SharedPreferenceUpdater() {
+                override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                    return when (newValue) {
+                        true -> {
+                            context.components.core.pocketStoriesService.startPeriodicSponsoredStoriesRefresh()
+                            super.onPreferenceChange(preference, newValue)
+                        }
+                        false -> {
+                            preference.isEnabled = false
+
+                            requireActivity().lifecycleScope.launch(Main) {
+                                val wasProfileDeletionSuccessful = withContext(ioDispatcher) {
+                                    context.components.core.pocketStoriesService.deleteProfile()
+                                }
```

Comment:
I'm not familiar with the internals of Pocket. what's the TLDR of what's happening here/why it's necessary?